src/custom-snippets.ts

```diff
@@ -94,21 +94,23 @@ async function evaluateCodeOrKey(codeOrKey?: string) {
   }
   if (pick === undefined) {
     // still no pick, but codeOrKey might be one
-    pick = snippetsKeyDict[codeOrKey];
+    pick = snippetsKeyMap.get(codeOrKey);
   }
-  const code = pick !== undefined ? snippetsDict[pick].snippet : codeOrKey;
-  const ns = pick !== undefined ? snippetsDict[pick].ns : editorNS;
-  const repl = pick !== undefined ? snippetsDict[pick].repl : editorRepl;
+
+  const replSnippet = snippetsMap.get(pick);
+
+  const code = pick !== undefined ? replSnippet?.snippet : codeOrKey;
+  const ns = pick !== undefined ? replSnippet?.ns : editorNS;
+  const repl = pick !== undefined ? replSnippet?.repl : editorRepl;
 
   const options = {};
 
   if (pick !== undefined) {
-    options['evaluationSendCodeToOutputWindow'] =
-      snippetsDict[pick].evaluationSendCodeToOutputWindow;
+    options['evaluationSendCodeToOutputWindow'] = replSnippet?.evaluationSendCodeToOutputWindow;
```

Comment:
strange note: `evaluationSendCodeToOutputWindow` was missing from `customREPLCommandSnippet` and I had to add it.